e2e/machine_migration_test.go

```diff
@@ -100,17 +119,208 @@ var _ = Describe("[sig-cluster-lifecycle][OCPFeatureGate:MachineAPIMigration] Ma
 			})
 
 			It("should verify that the non-authoritative MAPI Machine has an authoritative CAPI Machine mirror", func() {
-				Eventually(func() error {
-					newCapiMachine, err = capiframework.GetMachine(cl, mapiMachineAuthCAPIName, capiframework.CAPINamespace)
-					return err
-				}, capiframework.WaitMedium, capiframework.RetryMedium).Should(Succeed(), "CAPI Machine should exist")
+				newCapiMachine, err = capiframework.GetMachine(cl, mapiMachineAuthCAPIName, capiframework.CAPINamespace)
+				Expect(err).ToNot(HaveOccurred(), "CAPI Machine should exist")
 			})
 
 			It("should verify CAPI Machine Paused condition is False", func() {
 				verifyCAPIMachinePausedCondition(newCapiMachine, machinev1beta1.MachineAuthorityClusterAPI)
 			})
 		})
 	})
+
+	var _ = Describe("Update MAPI Machine", Ordered, func() {
+		Context("with spec.authoritativeAPI modification", func() {
+			It("should allow modification of authoritativeAPI from ClusterAPI to MachineAPI", func() {
+				By("Attempting to modify authoritativeAPI from ClusterAPI to MachineAPI")
+
+				// Create a new machine for this test to avoid affecting other tests
+				testMachineName := "machine-auth-change-83955"
+				testMachine := createMAPIMachineWithAuthority(ctx, cl, testMachineName, machinev1beta1.MachineAuthorityClusterAPI)
+
+				// Wait for the CAPI machine to be created (any status is fine for this test)
+				var err error
+				_, err = capiframework.GetMachine(cl, testMachineName, capiframework.CAPINamespace)
+				Expect(err).ToNot(HaveOccurred(), "CAPI Machine should be created")
+
+				// Add a small delay to ensure any pending updates are processed
+				Eventually(func() error {
+					// Get the latest version again right before update
+					currentMachine, err := mapiframework.GetMachine(cl, testMachineName)
+					if err != nil {
+						return err
+					}
+
+					updatedMachine := currentMachine.DeepCopy()
+					updatedMachine.Spec.AuthoritativeAPI = machinev1beta1.MachineAuthorityMachineAPI
+
+					err = cl.Update(ctx, updatedMachine)
+					return err
+				}, capiframework.WaitMedium, capiframework.RetryMedium).Should(Succeed(), "Should allow modification of authoritativeAPI")
+
+				// Verify the machine still exists after the change (don't wait for specific status)
+				_, err = mapiframework.GetMachine(cl, testMachineName)
+				Expect(err).ToNot(HaveOccurred(), "Machine should still exist after authoritativeAPI change")
```

Comment:
```suggestion
				Expect(mapiframework.GetMachine(cl, testMachineName)).To(Succeed(), "Machine should still exist after authoritativeAPI change")
```

Reply:
@theo , I get below error if I use the above suggestion probably the reason is function returns two values inside Expect  .

Reply:
Expected an error-type.  Got:
    <*v1beta1.Machine | 0xc000ad8008>: